### PR TITLE
Update Terraform plugin sdk v2

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## WHAT

Updated

* Context functions
* Add description to all resources

## WHY

* To maintain

## Ref

* [Terraform Plugin SDK v2 Upgrade Guide](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html)

### CODE REVIEW

**Comment by KeisukeYamashita** [comment]
@micnncim @dragon3 @Nessex 
Any updates?

File: main.go [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
+			&plugin.ServeOpts{
+				ProviderFunc: spinnaker.Provider,
+			})
+		if err != nil {
+			log.Println(err.Error())
+		}
+	} else {
+		plugin.Serve(&plugin.ServeOpts{
+			ProviderFunc: spinnaker.Provider})
+	}
```
Comment (dragon3) [inline main.go:29]: [note]
This is new implementation for Debuggable Provider Binary:
https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-debuggable-provider-binaries

File: spinnaker/data_source_spinnaker_application.go [hunk 2]
```diff
@@ -35,6 +36,6 @@ func datasourceApplication() *schema.Resource {
 				Computed: true,
 			},
 		},
-		Read: resourceSpinnakerProjectRead,
+		ReadContext: resourceSpinnakerApplicationRead,
```
Comment (dragon3) [inline spinnaker/data_source_spinnaker_application.go:39]: Just question, the previous implementation was incorrect, right? ( It should have been `Application`, not `Project` )

File: spinnaker/data_source_spinnaker_canary_config.go [hunk 3]
```diff
@@ -45,6 +46,6 @@ func datasourceCanaryConfig() *schema.Resource {
 				},
 			},
 		},
-		Read: resourceSpinnakerCanaryConfigRead,
+		ReadContext: resourceSpinnakerCanaryConfigRead,
```
Comment (dragon3) [inline spinnaker/data_source_spinnaker_canary_config.go:49]: [note]
[More Support for context.Context](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#more-support-for-context-context)

>  for example, the helper/schema.CreateFunc, helper/schema.UpdateFunc, helper/schema.ReadFunc, and helper/schema.DeleteFunc types all got context-aware versions (helper/schema.CreateContextFunc, helper/schema.UpdateContextFunc, helper/schema.ReadContextFunc, helper/schema.DeleteContextFunc, respectively). The old versions are now deprecated, and we recommend you upgrade to the context-aware versions to get their improvements, ...

File: spinnaker/provider_test.go [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
 	"testing"
 
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-var testAccProviders map[string]terraform.ResourceProvider
+var testAccProviders map[string]*schema.Provider
```
Comment (dragon3) [inline spinnaker/provider_test.go:12]: [note]
[Removal of the terraform.ResourceProvider Interface](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#removal-of-the-terraform-resourceprovider-interface)

> The terraform.ResourceProvider interface has been removed. The concrete *helper/schema.Provider type should be used in its place.

File: spinnaker/resource_application.go [hunk 5]
```diff
@@ -88,32 +91,34 @@ type Permissions struct {
 	Write   []string `json:"WRITE"`
 }
 
-func resourceSpinnakerApplicationCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceSpinnakerApplicationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (dragon3) [inline spinnaker/resource_application.go:94]: [note]
[Support for Diagnostics](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-diagnostics)

> A number of new functions have been added that return a diag.Diagnostics type. This type can be used to return multiple errors and warnings to Terraform, and to associate those errors or warnings with specific fields.

File: spinnaker/resource_application.go [hunk 6]
```diff
@@ -143,66 +148,46 @@ func resourceSpinnakerApplicationRead(d *schema.ResourceData, meta interface{})
...
 	if err := api.DeleteApplication(client, appName); err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	d.SetId("")
-	return nil
-}
-
-func resourceSpinnakerApplicationExists(d *schema.ResourceData, meta interface{}) (bool, error) {
```
Comment (dragon3) [inline spinnaker/resource_application.go:182]: [note]
[Deprecation of helper/schema.ExistsFunc](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#deprecation-of-helper-schema-existsfunc)

> The helper/schema.ExistsFunc (usually set as the Exists property on helper/schema.Resource) is now deprecated.

**Review by dragon3 — APPROVED** [review]
Looks good to me 👍 
( I added some notes for me 😬 and I'd like to upgrade the SDK for https://github.com/mercari/terraform-provider-openpgp as well )

File: spinnaker/data_source_spinnaker_application.go (hunk 2 above)
Comment (KeisukeYamashita) [reply spinnaker/data_source_spinnaker_application.go:39]: Yes, it was incorrect 🙇

**Comment by KeisukeYamashita** [comment]
Thank you for your review 🙇

**Pull request merged**